numpy/core/src/multiarray/datetime.c

```diff
@@ -1622,6 +1622,15 @@ compute_datetime_metadata_greatest_common_divisor(
 
     return 0;
 
+    /*
+     * The following errors do not currently use "DTypePromotionError".  This
+     * could be replaced (changing the OverflowError, though) with an explicit
+     * promotion error.
+     * This makes sense if we `arr_dt1 == arr_dt2` can reasonably return
```

Comment:
"if we" -> "since"?

I have a bit of trouble following the comment. Maybe easier to write it as
```
It might make sense to replace the following errors with "DTypePromotionError",
since we are here unable to convert one dtype to another.

But one would have to think of whether one then wants `arr_dt1 == arr_dt2` to return
an array of all `False`.
```
Aside: I don't think it should return an array of all `False` for either case, since it is just that
the conversion is difficult, not that it is not possible in principle. And some of the times
might well actually be equal (e.g., if they are 0?). So, I think an error is actually better.

Reply:
Ah, good point about 0, I think that is a killer argument (until we wish to change the `==` implementation, but even then I think the argument makes sense).